### PR TITLE
Update dcos-launch config to use onprem

### PR DESCRIPTION
**Pull request by cprovencher**
attempt to fix ~10% cluster deployment failure rate

### CODE REVIEW

File: ci/launch_cluster.sh [hunk 1]
```diff
@@ -38,16 +38,24 @@ echo "Using: ${TEMPLATE}"
 envsubst <<EOF > "$CONFIG_PATH"
 ---
 launch_config_version: 1
-template_url: $TEMPLATE
```
Comment (jeschkies) [inline ci/launch_cluster.sh]: Where did this go? We use two specific DC/OS pull requests:
* https://github.com/dcos/dcos/pull/1739 for DC/OS with latest Marathon
* https://github.com/mesosphere/dcos-enterprise/pull/1296 for DC/OS EE with latest Marathon

File: ci/launch_cluster.sh [hunk 2]
```diff
@@ -38,16 +38,24 @@ echo "Using: ${TEMPLATE}"
...
 deployment_name: $DEPLOYMENT_NAME
-provider: aws
+installer_url: https://downloads.dcos.io/dcos/testing/master/dcos_generate_config.sh
```
Comment (jeschkies) [inline ci/launch_cluster.sh]: Should this just be `https://downloads.dcos.io/dcos/$CHANNEL/dcos_generate_config.sh`?

**Comment by jeschkies** [comment]
Hm, the cluster launch fails https://jenkins.mesosphere.com/service/jenkins/job/system-integration-tests/job/marathon-si-pr/192/ :/

File: ci/launch_cluster.sh [hunk 3]
```diff
@@ -38,16 +38,24 @@ echo "Using: ${TEMPLATE}"
...
 key_helper: true
-template_parameters:
-    DefaultInstanceType: m4.large
-    AdminLocation: 0.0.0.0/0
-    PublicSlaveInstanceCount: 1
-    SlaveInstanceCount: 3
+instance_type: m4.large
+num_public_agents: 1
+num_private_agents: 3
+num_masters: 1
```
Comment (jeschkies) [inline ci/launch_cluster.sh]: This is also a multi master setup so we need 3.

File: ci/launch_cluster.sh (hunk 2 above)
Comment (MrMarvin) [reply ci/launch_cluster.sh]: Yep, should be something like
```
https://downloads.mesosphere.com/dcos-enterprise/testing/pull/1739/dcos_generate_config.ee.sh
```
where `testing/pull/1739` is your `$CHANNEL` var.

Comment (zen-dog) [reply ci/launch_cluster.sh]: yep, done

**Comment by zen-dog** [comment]
@cprovencher I've updated your branch so don't forget to pull

**Comment by cprovencher** [comment]
This same config was working when using dcos-launch in a different AWS account. AWS stack cannot come up, Will require more investigation.

File: ci/launch_cluster.sh [hunk 4]
```diff
@@ -26,28 +26,35 @@ CONFIG_PATH="$DEPLOYMENT_NAME.yaml"
 INFO_PATH="$DEPLOYMENT_NAME.info.json"
 
 if [ "$VARIANT" == "open" ]; then
-  TEMPLATE="https://s3.amazonaws.com/downloads.dcos.io/dcos/${CHANNEL}/cloudformation/multi-master.cloudformation.json"
+  INSTALLER="https://downloads.mesosphere.com/dcos/${CHANNEL}/dcos_generate_config.sh"
```
Comment (ichernetsky) [inline ci/launch_cluster.sh]: Why using CDN and not just S3 directly?

Comment (zen-dog) [reply ci/launch_cluster.sh]: Recommendations of the SRE team? ¯\_(ツ)_/¯

**Comment by zen-dog** [comment]
> This same config was working when using dcos-launch in a different AWS account. AWS stack cannot come up, Will require more investigation.

Here is an example of the `CREATE_FAILED` error: https://jenkins.mesosphere.com/service/jenkins/view/Marathon/job/system-integration-tests/job/marathon-si-pr/196/console

**Comment by MrMarvin** [comment]
Hey Folks, it seems there has been some confusion about which URL to use. Also I made a mistake by suggesting an enterprise URL for the open-DC/OS case earlier, apologies!

Below are valid URL templates for both cases:
```
if [ "$VARIANT" == "open" ]; then
  INSTALLER="https://downloads.dcos.io/dcos/${CHANNEL}/dcos_generate_config.sh"
else
  INSTALLER="https://downloads.mesosphere.com/dcos-enterprise/${CHANNEL}/dcos_generate_config.ee.sh"
fi
```

Please also always try to use the CDN URLs, as they allow us to abstract from individual bucket names, which will likely change at some point.

**Comment by cprovencher** [comment]
@zen-dog I believe I have found the culprit: m5.Large instance type. I don't think those are supported by the current templates we have in dcos-launch. m4 instances work fine. The easy solution is to switch to m4 instance types, but if you really need m5 we'll have to examine the templates and add support for m5 instances https://github.com/dcos/dcos-launch/tree/master/dcos_launch/templates

**Comment by jeschkies** [comment]
Ok, the cluster started at least. However, we don't have the ssh key.

**Comment by jeschkies** [comment]
Only `test_marathon_root.test_external_volume` is failing. It uses rexray. Do we know of any rexray limitations in CentOS?

**Comment by jeschkies** [comment]
Strict hat two failures in two runs
- `test_marathon_root.test_launch_mesos_container_with_docker_image`
- `test_marathon_root.test_private_repository_docker_app`

Both seem falkes to me.

File: ci/launch_cluster.sh [hunk 5]
```diff
@@ -26,33 +26,43 @@ CONFIG_PATH="$DEPLOYMENT_NAME.yaml"
 INFO_PATH="$DEPLOYMENT_NAME.info.json"
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/launch_cluster.sh]: The iterface to `ci/launch_cluster.sh` did not change. So it should be fine...in theory 😉.

**Pull request merged**